turbopack/crates/turbopack-ecmascript/src/tree_shake/optimizations.rs

```diff
@@ -246,4 +246,205 @@ impl GraphOptimizer<'_> {
 
         did_work
     }
+
+    /// If the graph is
+    ///
+    /// a -> c -> d -> f
+    /// a -> c -> e -> f
+    /// b -> c
+    ///
+    /// all of `d, e, f` should be merged into `c` because they are all reachable only by stepping
+    /// through `c`.
+    pub(super) fn perform_dominator_analysis<N>(&self, g: &mut Graph<Vec<N>, Dependency>) -> bool
```

Comment:
do you have an estimate of the runtime complexity of this function?  could you document it?